### PR TITLE
Increment default python version to 3.11

### PR DESCRIPTION
**Pull request by jmsanders**
I suspect we'll want to sufficiently increment the released version so this doesn't break users who are pinned to 0.1.

### CODE REVIEW

File: gitlab/serverless-legacy-ci.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ build-image:
   before_script:
     - echo $AWS_ECR_PASSWORD | docker login --username $AWS_ECR_USERNAME --password-stdin $REGISTRY_URL
   script:
-    - echo "FROM python:3.8-slim" > $DAGSTER_CLOUD_LOCATION_DIR/Dockerfile
+    - echo "FROM python:3.11-slim" > $DAGSTER_CLOUD_LOCATION_DIR/Dockerfile
```
Comment (jmsanders) [inline gitlab/serverless-legacy-ci.yml:43]: Given the name, I'm not sure if we should be touching this one?

**Review by gibsondan — COMMENTED** [review]
As per discussion on slack i think this would require a v0.2 release

we should also update the build/release instructions to clarify that it needs to be done from a 3.11 venv (it currently needs to be done from a 3.8 venv or the resulting built hybrid CI action fails)

**Comment by shalabhc** [comment]
I think we want to change only the templates that get copied into the user repo. Eg https://github.com/dagster-io/dagster-cloud-action/tree/main/github/serverless

**Comment by jmsanders** [comment]
So keep all the defaults as 3.8 but have all the user templates override to 3.11? It looks like we might need to add a python_version override to the gitlab templates in that case.

**Review by gibsondan — COMMENTED** [review]
i'm unclear where exactly this landed - does this now only apply to newly templated github actions? If so (and we've double-checked that the regular example and dbt nux work out of hte box for 3.11) then I would support landing thsi

**Pull request closed without merging**